### ISSUE TITLE
Support Cross Namespace PoolObjectReferences

### ISSUE THREAD

**Issue by danehans** [kubernetes-sigs/gateway-api-inference-extension#113]
Since InferencePool and InferenceModel resources are managed by different personas they may reside in different namespaces. Therefore, PoolObjectReference should support an optional `namespace` field that defines the namespace of the referenced object. When this field is unspecified or specifies the same namespace as the InferencePool, the referenced object must be in the same namespace as the InferencePool. When `namespace` specifies a different namespace from the InferencePool, a ReferenceGrant object is required in the referenced object's namespace to allow the reference.

**Comment by ahg-g** [comment]
I discussed this with @robscott briefly in the initial API proposal and we opted to start without allowing cross namespace referencing of resources to avoid dealing with that complexity and not to have dependencies on the upstream gateway apis initially. @robscott wdyt?

With ReferenceGrant, is it the extension itself that needs to honor the policy?

**Comment by danehans** [comment]
+1 on not supporting this issue for now but let's keep it open to track interest.

**Comment by danehans** [comment]
xref ttps://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/149#issuecomment-2574126971 from @liu-cong

**Comment by danehans** [comment]
@ahg-g @robscott I understand that this issue is related to an InferenceModel -> InferencePool reference, but should the same apply to an HTTPRoute -> InferencePool reference? If not, we should add a conformance test for an HTTPRoute -> InferencePool cross-ns reference.

**Comment by robscott** [comment]
> [@ahg-g](https://github.com/ahg-g) [@robscott](https://github.com/robscott) I understand that this issue is related to an InferenceModel -> InferencePool reference, but should the same apply to an HTTPRoute -> InferencePool reference? If not, we should add a conformance test for an HTTPRoute -> InferencePool cross-ns reference.

Yeah, I think it makes sense to support ReferenceGrant for HTTPRoute -> InferencePool, just like any other backend. Agree that we should have a conformance test for this.
